mlrun/common/formatters/model_endpoint.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import typing
+
+import mlrun.common.types
+
+from .base import ObjectFormat
+
+
+class ModelEndpointFormat(ObjectFormat, mlrun.common.types.StrEnum):
+    minimal = "minimal"
```

Comment:
You should also include the `full` format here, if not you will get an error:
`value is not a valid enumeration member; permitted: 'minimal' (type=type_error.enum; enum_values=[minimal])`

ModelEndpointFormat  inherits from both ObjectFormat and StrEnum, which can create confusion with the enumeration members. In Python, enum.Enum typically expects all members to be defined at the same level in the class. Mixing members from different parent classes can lead to validation issues. The solution is to define all enum members in one place.

Reply:
thanks!